.github/workflows/misc-migtests.yml

```diff
@@ -43,6 +43,14 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-maven-
 
+      - name: Cache Go modules
+        uses: actions/cache@v3
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
```

Comment:
Just curious, how much benefit does this provide?
Since we might want to replicate what our users face while installing Voyager.

Reply:
It doesn't seem to be providing any speedup. I thought to just keep it thinking it won't harm anything.
Are you suggesting to remove it?

Reply:
Yes if there's no benefit, we should remove it. Voyager installation should handle it to replicate the user's environment as closely as possible.

Reply:
Will remove it. 
Btw it shouldn't make an difference wrt to replicating user's environment. If the `go.sum`(checksum) files for dependent modules is same then it would cache them and not install again. 
Voyager will be build irrespective of that.

Reply:
Ah right! thanks :)